doc/stdenv/stdenv.chapter.md

```diff
@@ -991,13 +991,45 @@ Hook executed at the end of the fixup phase.
 
 If set to `true`, the standard environment will enable debug information in C/C++ builds. After installation, the debug information will be separated from the executables and stored in the output named `debug`. (This output is enabled automatically; you don’t need to set the `outputs` attribute explicitly.) To be precise, the debug information is stored in `debug/lib/debug/.build-id/XX/YYYY…`, where \<XXYYYY…\> is the \<build ID\> of the binary — a SHA-1 hash of the contents of the binary. Debuggers like GDB use the build ID to look up the separated debug information.
 
-For example, with GDB, you can add
+For example, to make GDB able to find debug information for the `socat` package and its dependencies, you can use the following
+`shell.nix`:
+
+```.nix
+let
+  pkgs = import ./. {
+    config = {};
+    overlays = [
+      (final: prev: {
+        ncurses = prev.ncurses.overrideAttrs { separateDebugInfo = true; };
+        readline = prev.readline.overrideAttrs { separateDebugInfo = true; };
+      })
+    ];
+  };
 
-```
-set debug-file-directory ~/.nix-profile/lib/debug
-```
+  myDebugInfoDirs = pkgs.symlinkJoin {
+    name = "myDebugInfoDirs";
+    paths = [
+      pkgs.glibc.debug
+      pkgs.ncurses.debug
+      pkgs.openssl.debug
+      pkgs.readline.debug
+    ];
+  };
+in
+  pkgs.mkShell {
 
-to `~/.gdbinit`. GDB will then be able to find debug information installed via `nix-env -i`.
+    NIX_DEBUG_INFO_DIRS = "${pkgs.lib.getLib myDebugInfoDirs}/lib/debug";
+
+    packages = [
+      pkgs.gdb
+      pkgs.socat
+    ];
+
+    shellHook = ''
+      ${pkgs.lib.getBin pkgs.gdb}/bin/gdb ${pkgs.lib.getBin pkgs.socat}/bin/socat
+    '';
+  }
+```
```

Comment:
````suggestion
```
:::
````

Reply:
We may want to add some explanation to this, linking to the relevant concepts. It's quite likely someone will want to do debugging before they know all the details about Nixpkgs, so they could navigate from here:

```
This setup works as follows:
- Add [overlays] to the package set, since debug symbols are disabled for `ncurses` and `readline` by default.
- Create a derivation to combine all required debug symbols under one path with [`symlinkJoin`].
- Set the environment variable `NIX_DEBUG_INFO_DIRS` in the shell. Nixpkgs patches `gdb` to use it for looking up debug symbols.
- Run `gdb` on the `socat` binary on shell startup in the [`shellHook`]. Here we use [`lib.getBin`] to ensure that the correct derivation output is selected rather than the default one.
```